### PR TITLE
Folding Chairs

### PR DESCRIPTION
**Pull request by Shadowmech88**
south east north west
Adds folding chairs, which can be crafted from 1 sheet of metal.
They function like regular chairs, but can be folded up into an item via click+dragging them onto yourself, similar to folding a roller bed.
Folding chairs are also not anchored.

:cl:
 * rscadd: Added folding chairs, which can be constructed with 1 sheet of metal. Folding chairs can be folded up by click+dragging them onto yourself, similar to roller beds.

### CODE REVIEW

**Comment by SonixApache** [comment]
do they fall over if you drag them

**Comment by Shadowmech88** [comment]
They don't, but neither do actual folding chairs.

**Comment by Exxion** [comment]
Why all the creation and deletion? Just store the chair in the item and maybe even also store the item in the chair

**Comment by Shadowmech88** [comment]
I just did it the way roller beds do it. I can change it, though.

**Comment by Exxion** [comment]
Wow I'm going to meme on whoever created roller beds

**Comment by JustSumBody** [comment]
image

**Comment by PJB3005** [comment]
Dear God that's an awful sprite.

**Comment by cadaverouscoyote** [comment]
yeah the sprite's pretty bad. the perspective's completely different from other chairs, is it from another server or something?

**Comment by Shadowmech88** [comment]
I took them from #7992 which apparently took them from Hippie, and edited them a bit.

**Comment by cadaverouscoyote** [comment]
man i forgot about those ugly things

**Comment by ghost** [comment]
Post a screenshot of them in-game please

**Comment by Shadowmech88** [comment]
image

**Comment by ghost** [comment]
ok yeah personally I would have posted one not against a grey floor

**Comment by Gunosaguki** [comment]
Do they do bonus damage when thrown?

**Comment by Shadowmech88** [comment]
image

**Comment by ghost** [comment]
I'm not 💯 on the sprite but it's not TERRIBLE either and I'm excited for the robust opportunities this provides.

**Comment by Riekland** [comment]
Doesn't goon have the same chair sprites as us but they also have a folded one
just use theirs
actually I think they might have changed theirs, I know they had the same as us like a year ago or something go look then

**Comment by PJB3005** [comment]
Goon replaced their regular chair sprite with really good folding chairs IIRC.

**Comment by moistnuggie** [comment]
can we fold them and use them as wrestling cage weapons?

**Comment by moistnuggie** [comment]
can we fold them and use them as wrestling weapons?

**Comment by salamandooo** [comment]
Very good for beach parties

**Comment by Riekland** [comment]
I thought this wasn't getting merged until it didn't look like aids what to fuck

**Comment by PJB3005** [comment]
Want me back now don't you.

**Comment by ghost** [comment]
???

**Comment by Riekland** [comment]
I want pomf back

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 1]
```diff
@@ -405,3 +405,70 @@
...
+
+//Folding chair
+
+/obj/structure/bed/chair/folding
+	name = "folding chair"
+	icon_state = "folding_chair"
+	anchored = 0
+	var/obj/item/folding_chair/folded
+
+/obj/structure/bed/chair/folding/New(turf/T, var/chair)
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:417]: The `chair` parameter here should *really* have a type.

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 2]
```diff
@@ -405,3 +405,70 @@
...
+	..(T)
+	if(!folded)
+		if(chair)
+			folded = chair
+		else
+			folded = new(src, src)
+
+/obj/structure/bed/chair/folding/Destroy()
+	if(folded)
+		folded.unfolded = null
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:427]: This should be done in `folded`'s `Destroy` instead.

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 3]
```diff
@@ -405,3 +405,70 @@
...
+	..()
+
+/obj/item/folding_chair/attack_self(mob/user)
+	unfolded.forceMove(user.loc)
+	unfolded.add_fingerprint(user)
+	unfolded.dir = user.dir
+	user.drop_item(src, force_drop = 1)
+	forceMove(unfolded)
+
+/obj/structure/bed/chair/folding/MouseDrop(over_object, src_location, over_location)
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:462]: Move this to before the item definition.

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 4]
```diff
@@ -405,3 +405,70 @@
...
+		unfolded.folded = null
+		qdel(unfolded)
+		unfolded = null
+	..()
+
+/obj/item/folding_chair/attack_self(mob/user)
+	unfolded.forceMove(user.loc)
+	unfolded.add_fingerprint(user)
+	unfolded.dir = user.dir
+	user.drop_item(src, force_drop = 1)
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:459]: How exactly are you putting a chair down if it's stuck to your hand?

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 5]
```diff
@@ -405,3 +405,70 @@
...
+	if(unfolded)
+		unfolded.folded = null
+		qdel(unfolded)
+		unfolded = null
+	..()
+
+/obj/item/folding_chair/attack_self(mob/user)
+	unfolded.forceMove(user.loc)
+	unfolded.add_fingerprint(user)
+	unfolded.dir = user.dir
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:458]: Should use `change_dir`

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 6]
```diff
@@ -405,3 +405,70 @@
...
+
+/obj/item/folding_chair
+	name = "folding chair"
+	desc = "A collapsed folding chair that can be carried around."
+	icon = 'icons/obj/stools-chairs-beds.dmi'
+	icon_state = "folded_chair"
+	w_class = W_CLASS_LARGE
+	var/obj/structure/bed/chair/folding/unfolded
+
+/obj/item/folding_chair/New(turf/T, var/chair)
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:440]: Type here.

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 7]
```diff
@@ -405,3 +405,70 @@
...
+	forceMove(unfolded)
+
+/obj/structure/bed/chair/folding/MouseDrop(over_object, src_location, over_location)
+	..()
+	if(over_object == usr && Adjacent(usr))
+		if(!ishuman(usr) || usr.incapacitated() || usr.lying)
+			return
+
+		if(is_locking(lock_type))
+			return 0
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:469]: Really inconsistent return values here.

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 8]
```diff
@@ -405,3 +405,70 @@
...
+	unfolded.add_fingerprint(user)
+	unfolded.dir = user.dir
+	user.drop_item(src, force_drop = 1)
+	forceMove(unfolded)
+
+/obj/structure/bed/chair/folding/MouseDrop(over_object, src_location, over_location)
+	..()
+	if(over_object == usr && Adjacent(usr))
+		if(!ishuman(usr) || usr.incapacitated() || usr.lying)
+			return
```
Comment (PJB3005) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:466]: Can be part of the if above.

**Review by PJB3005 — COMMENTED** [review]
Nice.

**Comment by FantasticFwoosh** [comment]
On TG we just pick up chairs/stools etc outright, but this is pretty cool eitherway, a nice item to throw into the rage cage.

**Pull request merged**